src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkBaseExternalContext.java

```diff
@@ -1352,21 +1352,17 @@ during the analysis phase and thus cannot depends on a target result (the \
                     + " a path from.")
       })
   public StarlarkPath path(Object path) throws EvalException, InterruptedException {
-    return getPath("path()", path);
+    return getPath(path);
   }
 
-  protected StarlarkPath getPath(String method, Object path)
-      throws EvalException, InterruptedException {
-    if (path instanceof String) {
-      return new StarlarkPath(this, workingDirectory.getRelative(path.toString()));
-    } else if (path instanceof Label label) {
-      return getPathFromLabel(label);
-    } else if (path instanceof StarlarkPath starlarkPath) {
-      return starlarkPath;
-    } else {
+  protected StarlarkPath getPath(Object path) throws EvalException, InterruptedException {
+    return switch (path) {
+      case String ignored -> new StarlarkPath(this, workingDirectory.getRelative(path.toString()));
```

Comment:
```suggestion
      case String s -> new StarlarkPath(this, workingDirectory.getRelative(s));
```

less surprising this way, I think ("why the ignore?")